### PR TITLE
Add sshcert auth

### PR DESCRIPTION
**Pull request by cviecco**
This is the server side part of the sshcert auth portion. Only minimal test have been added. Notice that this is against the devel branch, which I just created so that we can start having common development here... I think the master branch is just getting too busy.

### CODE REVIEW

File: cmd/keymasterd/app.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ import (
 	"github.com/Cloud-Foundations/tricorder/go/tricorder"
 	"github.com/Cloud-Foundations/tricorder/go/tricorder/units"
 	"github.com/cloudflare/cfssl/revoke"
+	"github.com/cviecco/webauth-sshcert/lib/server/sshcertauth"
```
Comment (rgooch) [inline cmd/keymasterd/app.go:60]: :cry:

File: cmd/keymasterd/app.go [hunk 2]
```diff
@@ -1937,6 +1950,12 @@ func main() {
 		serviceMux.HandleFunc(paths.VerifyAuthToken,
 			runtimeState.VerifyAuthTokenHandler)
 	}
+	// TODO: only enable these handlers if sshcertauth is enabled
```
Comment (rgooch) [inline cmd/keymasterd/app.go:1953]: When will this be done?

File: cmd/keymasterd/auth_sshcert.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/Cloud-Foundations/keymaster/lib/webapi/v0/proto"
+	"golang.org/x/crypto/ssh"
+)
+
+// Thus function can only be called after all known keymaster public keys
```
Comment (rgooch) [inline cmd/keymasterd/auth_sshcert.go:11]: s/Thus/This/

File: cmd/keymasterd/auth_sshcert.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		// we are assuming bad request
+		s.logger.Debugf(1,
+			"CreateSSHCertAuthChallengeHandler: there was an err computing challenge: %s", err)
+		s.writeFailureResponse(w, r, http.StatusBadRequest, "Invalid Operation")
+		return
+	}
+}
+
+func (s *RuntimeState) sshCertAuthLoginWithChallengeHandler(w http.ResponseWriter, r *http.Request) {
+	username, maxAge, userErrString, err := s.sshCertAuthenticator.LoginWithChallenge(r)
```
Comment (rgooch) [inline cmd/keymasterd/auth_sshcert.go:43]: s/maxAge/expiration/
Apply throughout function.

**Review by rgooch — CHANGES_REQUESTED** [review]

File: cmd/keymasterd/app.go (hunk 2 above)
Comment (cviecco) [reply cmd/keymasterd/app.go:1959]: done

File: cmd/keymasterd/auth_sshcert.go (hunk 3 above)
Comment (rgooch) [reply cmd/keymasterd/auth_sshcert.go:11]: Ping?

File: cmd/keymasterd/auth_sshcert.go (hunk 4 above)
Comment (rgooch) [reply cmd/keymasterd/auth_sshcert.go:52]: Ping?

**Review by rgooch — CHANGES_REQUESTED** [review]

File: cmd/keymasterd/auth_sshcert.go (hunk 3 above)
Comment (cviecco) [reply cmd/keymasterd/auth_sshcert.go]: done

**Review by rgooch — APPROVED** [review]

**Pull request merged**